presto-hive/src/main/java/com/facebook/presto/hive/util/InternalHiveSplitFactory.java

```diff
@@ -240,14 +239,7 @@ private static boolean hasRealAddress(List<HostAddress> addresses)
 
     private static List<HostAddress> getHostAddresses(BlockLocation blockLocation)
     {
-        String[] hosts;
-        try {
-            hosts = blockLocation.getHosts();
-        }
-        catch (IOException e) {
```

Comment:
Changing from UncheckedIOException to IOException is intentional?

Reply:
I guess so. I didnt change the behavior. Its the same as before my PR

Reply:
Oh you are right, the exception throw logic is moved to BlockLocation.